update kind to 0.20.0

kannon92 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind cleanup
/kind testing
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:
Update Kind to 0.20.0 so we can pick up 1.29.0 images.

Drop deprecated kubernetes versions.
#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:
I am noticing something odd in Kind 0.20.0 around cert rotation so I think we should test the CI pretty well here once 1.29 test-infra PR merges.

See https://github.com/kubernetes-sigs/jobset/issues/357 for some discussion on this.
#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

kannon92 commented:
/hold

kannon92 commented:
/test pull-kueue-test-e2e-main-1-28

tenzen-y commented on Makefile:
```diff
@@ -65,7 +65,7 @@ E2E_TARGET ?= ./test/e2e/...
 
 # E2E_K8S_VERSIONS sets the list of k8s versions included in test-e2e-all
-E2E_K8S_VERSIONS ?= 1.24.15 1.25.11 1.26.6 1.27.3 1.28.0
+E2E_K8S_VERSIONS ?= 1.26.6 1.27.3 1.28.0 1.29.0
```
@kannon92 Can we update the Kubernetes version in other places like `ENVTEST_K8S_VERSION`?

kannon92 replied:
Its at 1.28.0 and I don't know if we should update that to 1.29.0 yet.

alculquicondor replied:
let's use the latest patch version for 1.28

kannon92 replied:
updated. sorry for the lag.

kannon92 commented:
/test pull-kueue-test-e2e-main-1-28
/test pull-kueue-test-e2e-main-1-29

kannon92 commented:
@tenzen-y @alculquicondor I got a bit confused by kind's website. I was thinking that the latest versions of Kind would have 1.29 but it seems that 0.18.0 has it also.

Do you all still an update to the latest version of kind?

tenzen-y commented:
> @tenzen-y @alculquicondor I got a bit confused by kind's website. I was thinking that the latest versions of Kind would have 1.29 but it seems that 0.18.0 has it also.
> 
> Do you all still an update to the latest version of kind?

@kannon92 I see. Then, can we update only the KinD version in this PR?

kannon92 commented:
> > @tenzen-y @alculquicondor I got a bit confused by kind's website. I was thinking that the latest versions of Kind would have 1.29 but it seems that 0.18.0 has it also.
> > Do you all still an update to the latest version of kind?
> 
> @kannon92 I see. Then, can we update only the KinD version in this PR?

@alculquicondor and @tenzen-y I pushed a change for this and updated all the kind versions to be the latest kind and updated the version. It probably makes sense for kind image to be the latest patch.

tenzen-y reviewed (commented):
Thanks!
/lgtm
/apprvoe

tenzen-y commented:
/approve
